src/wallets/IdentityWallet.ts

```diff
@@ -43,16 +44,18 @@ export class IdentityWallet implements TLWallet {
   private identityAddress: string
   private identityFactoryAddress: string
   private identityImplementationAddress: string
+  private nonceMechanism: NonceMechanism
   private chainId: number
   private identityVersion = 1
 
   constructor(
     provider: TLProvider,
     chainId: number,
-    { identityFactoryAddress, identityImplementationAddress }
+    { identityFactoryAddress, identityImplementationAddress, nonceMechanism }
```

Comment:
You're right, I am not sure this way of writing inputs makes sense. I would rather have them not packed with proper typing

Reply:
:heavy_check_mark: